transport/errors.go

```diff
@@ -207,3 +210,32 @@ func (e procedureFailedError) Error() string {
 func (e procedureFailedError) AsHandlerError() HandlerError {
 	return LocalUnexpectedError(e)
 }
+
+//////////////////////////////////////////////////////////////////////////////
+
+type TimeoutError interface {
+	error
+
+	timeoutError()
+}
+
+type timeoutError struct {
+	Service   string
+	Procedure string
+	Duration  time.Duration
+}
+
+func NewTimeoutError(Service string, Procedure string, Duration time.Duration) timeoutError {
```

Comment:
This should return `error` or `TimeoutErorr`, not `timeoutError`. It's a public function, and `timeoutError` is a private type.

Also, `golint` and `go vet`